### ISSUE TITLE
Hyprpicker appears on the top left corner of my screen

### ISSUE THREAD

**Issue by ARKye03** [hyprwm/hyprpicker#85]
This is my system info:
<details>
    <summary>hyprctl systeminfo -c</summary>

```
Hyprland, built from branch main at commit 2b520571e897be2a0e88c8692da607b062000038  (keybinds: improve fullscreenstate toggling (7174)).
Date: Sun Aug 4 21:40:34 2024
Tag: v0.41.2-165-g2b520571, commits: 5051

flags: (if any)


System Information:
System name: Linux
Node name: archyalinux
Release: 6.10.2-zen1-2-zen
Version: #1 ZEN SMP PREEMPT_DYNAMIC Sat, 03 Aug 2024 18:22:59 +0000


GPU information: 
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation GA106 [Geforce RTX 3050] [10de:2507] (rev a1) (prog-if 00 [VGA controller])
NVRM version: NVIDIA UNIX x86_64 Kernel Module  555.58.02  Tue Jun 25 01:39:15 UTC 2024


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:

======Config-Start======


Config File: /home/archkye/.config/hypr/hyprland.conf: Read Succeeded
# See https://wiki.hyprland.org/Configuring/Monitors/
monitor=, 1920x1080, 0x0, 1
monitor=Unknown-1,disable

# Env vars
source = ~/.config/hypr/default/env.conf

input {
    kb_layout = us
    kb_variant = #intl 
    kb_model = 
    kb_options =
    kb_rules =
    numlock_by_default = true

    follow_mouse = 1

    touchpad {
        natural_scroll = false
    }

    sensitivity = 0 # -1.0 - 1.0, 0 means no modification.
}

general {
    gaps_in = 5
    gaps_out = 5
    border_size = 2
    col.active_border = rgba(8839efee) rgba(d20f39ee) 45deg
    col.inactive_border = rgba(595959aa)

    layout = dwindle
}

decoration {
    rounding = 5

    blur {
        enabled = true
        new_optimizations = true
        size = 5
        passes = 4
    }

    drop_shadow = false
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba(1a1a1aee)
}

animations {
    enabled = true

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05

    animation = windows, 1, 4, myBezier
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = borderangle, 1, 8, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 2, default
}

dwindle {
    pseudotile = true
    preserve_split = true
}

gestures {
    workspace_swipe = false
}

# Example per-device config
# See https://wiki.hyprland.org/Configuring/Keywords/#per-device-input-configs for more
device {
    name = epic-mouse-v1
    sensitivity = -0.5
}
#Autostart
source = ~/.config/hypr/default/startup.conf

#KeyBinds
source = ~/.config/hypr/default/binds.conf

#WholeWorldHere
source = ~/.config/hypr/default/windowrules.conf
source = ~/.config/hypr/default/layerrules.conf


Config File: /home/archkye/.config/hypr/default/env.conf: Read Succeeded
env = LIBVA_DRIVER_NAME,nvidia
env = XDG_SESSION_TYPE,wayland
env = GBM_BACKEND,nvidia-drm
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
env = NVD_BACKEND,direct
env = QT_QPA_PLATFORM,wayland
env = WLR_RENDERER_ALLOW_SOFTWARE,1
env = GDK_BACKEND,wayland
env = WARP_ENABLE_WAYLAND,1
env = ELECTRON_OZONE_PLATFORM_HINT,wayland


Config File: /home/archkye/.config/hypr/default/startup.conf: Read Succeeded
#Autostart
exec-once=~/.dotfiles/scripts/WBG.lua random
exec-once=/usr/bin/dbus-update-activation-environment --all
exec-once=/usr/bin/gnome-keyring-daemon --start --components=pkcs11,secrets,ssh
exec-once=/usr/bin/hyprctl setcursor "catppuccin-mocha-dark" 24
exec-once=/usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1
# exec-once=/usr/bin/mpd /home/archkye/.mpd/mpd.conf # Init via systemd
exec-once=/usr/bin/nm-applet
exec-once=/usr/bin/copyq --start-server
# exec-once=/usr/bin/nwg-dock-hyprland -d -nolauncher
exec-once=/usr/bin/hypridle
exec-once=/usr/bin/wl-clip-persist --clipboard regular
exec-once=/usr/bin/mpd-mpris --no-instance --host 127.0.0.1
exec-once=/usr/bin/ags # > ~/.zoore/ags.log 2>&1


Config File: /home/archkye/.config/hypr/default/binds.conf: Read Succeeded
$mainMod = SUPER

bind=$mainMod CTRL, Space, global, kando:master
# App binds
# Terms
bind=$mainMod, Return, exec, /usr/bin/alacritty
bind=$mainMod CTRL, Return, exec, /usr/bin/warp-terminal
bind=$mainMod SHIFT, Return, exec, /usr/bin/kitty

#Browser
bind=$mainMod, W, exec, /usr/bin/brave --enable-features=UseOzonePlatform --ozone-platform=wayland --use-gl=desktop --disable-features=UseChromeOSDirectVideoDecoder --enable-features=VaapiVideoDecoder

# Utils
bind= $mainMod, V, exec, ~/.dotfiles/scripts/toggle_vpn.lua
bind= $mainMod, X, exec, ~/.config/rofi/powermenu/type-4/powermenu.sh
bind= $mainMod, M, exec, /usr/bin/alacritty --class floatcritty -e /usr/bin/ncmpcpp -c ~/.ncmpcpp/config
bind= $mainMod, E, exec, /usr/bin/thunar

# Screenshots
bind= $mainMod, Print, exec, ~/.dotfiles/scripts/gss.lua region
bind= $mainMod, S, exec, ~/.dotfiles/scripts/gss.lua window
bind=, Print, exec, ~/.dotfiles/scripts/gss.lua full
bind= $mainMod, Q, exec, ~/.dotfiles/scripts/gss.lua aspect_visualizer

# Wallpapers WBG
bind= $mainMod, U, exec, ~/.dotfiles/scripts/WBG.lua next
bind= $mainMod SHIFT, U, exec, ~/.dotfiles/scripts/WBG.lua prev

# HyprEcko
bind= $mainMod, P, exec, /usr/bin/hyprpicker -a
bind= $mainMod, L, exec, /usr/bin/hyprlock

# BingGPT
bind= $mainMod, B, exec, /opt/microsoft/msedge/microsoft-edge --profile-directory=Default --app-id=hpgfdmhandgfpmaeglgcipnkmjepighl "--app-url=https://www.bing.com/search?toWww=1&redig=4BC86B9EF6E64CBEB46913C33D1FBEC4&q=Bing+AI&showconv=1"

#AGS
bind=CTRL SHIFT, R,  exec, /usr/bin/ags -q; ags # Restart ags
bindr=Super, D, exec, /usr/bin/ags -t applauncher
bindr=Super, A, exec, /usr/bin/ags -t powermenu
bind=CTRL SHIFT, G,  exec, /usr/bin/ags -t Chat 
# bind=$mainMod, Z,  exec, /usr/bin/ags -t cliphist

#WindowControl
bind= $mainMod, C, killactive,
bind= $mainMod, J, pseudo,
bind= $mainMod, K, togglesplit,
bind= $mainMod SHIFT, Q, exit,
bind= $mainMod, F, fullscreen,
bind= $mainMod, Space, togglefloating,

#Volume Keys
binde=, XF86AudioRaiseVolume, exec, /usr/bin/wpctl set-volume -l 1 @DEFAULT_AUDIO_SINK@ 5%+
binde=, XF86AudioLowerVolume, exec, /usr/bin/wpctl set-volume @DEFAULT_AUDIO_SINK@ 5%-

bind=, XF86AudioPlay, exec, /usr/bin/mpc toggle
bind=, XF86AudioNext, exec, /usr/bin/mpc next
bind=, XF86AudioPrev, exec, /usr/bin/mpc prev
bind=, XF86audiostop, exec, /usr/bin/mpc stop

# Move focus with mainMod + arrow keys
bind= $mainMod, left, movefocus, l
bind= $mainMod, right, movefocus, r
bind= $mainMod, up, movefocus, u
bind= $mainMod, down, movefocus, d

# Switch workspaces with mainMod + [0-9]
bind= $mainMod, 1, workspace, 1
bind= $mainMod, 2, workspace, 2
bind= $mainMod, 3, workspace, 3
bind= $mainMod, 4, workspace, 4
bind= $mainMod, 5, workspace, 5
bind= $mainMod, 6, workspace, 6
bind= $mainMod, 7, workspace, 7
bind= $mainMod, 8, workspace, 8
bind= $mainMod, 9, workspace, 9
bind= $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind= $mainMod SHIFT, 1, movetoworkspace, 1
bind= $mainMod SHIFT, 2, movetoworkspace, 2
bind= $mainMod SHIFT, 3, movetoworkspace, 3
bind= $mainMod SHIFT, 4, movetoworkspace, 4
bind= $mainMod SHIFT, 5, movetoworkspace, 5
bind= $mainMod SHIFT, 6, movetoworkspace, 6
bind= $mainMod SHIFT, 7, movetoworkspace, 7
bind= $mainMod SHIFT, 8, movetoworkspace, 8
bind= $mainMod SHIFT, 9, movetoworkspace, 9
bind= $mainMod SHIFT, 0, movetoworkspace, 10

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

bind = $mainMod CTRL, right, workspace, e+1
bind = $mainMod CTRL, left, workspace, e-1

bind = $mainMod ALT, right, movetoworkspace, r+1
bind = $mainMod ALT, left, movetoworkspace, r-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow

# sets repeatable binds for resizing the active window
binde=$mainMod SHIFT, right, resizeactive,10 0
binde=$mainMod SHIFT, left, resizeactive,-10 0
binde=$mainMod SHIFT, up, resizeactive,0 -10
binde=$mainMod SHIFT, down, resizeactive,0 10

# sets repeatable binds for resizing the active window
binde=$mainMod ALT SHIFT, right, moveactive,10 0
binde=$mainMod ALT SHIFT, left, moveactive,-10 0
binde=$mainMod ALT SHIFT, up, moveactive,0 -10
binde=$mainMod ALT SHIFT, down, moveactive,0 10

bind=$mainMod, TAB, workspace, previous
bind=$mainMod, GRAVE, togglespecialworkspace,
bind=$mainMod SHIFT, GRAVE, movetoworkspace, special
bind=$mainMod SHIFT, P, pin,


Config File: /home/archkye/.config/hypr/default/windowrules.conf: Read Succeeded
#--------Kando
windowrule = noblur, kando 
windowrule = size 100% 100%, kando
windowrule = noborder, kando
windowrule = noanim, kando
windowrule = float, kando
windowrule = pin, kando

#--------1
#windowrule = workspace 1, ^(.*Alacritty.*)$

# Brave
#--------2
windowrule = workspace 2, ^(.*brave-browser.*)$
#----Brave_PIP
windowrulev2 = float,title:(Picture in picture)
windowrulev2 = size 600 350,title:(Picture in picture)
windowrulev2 = move 1310 720,title:(Picture in picture)
windowrulev2 = pin,title:(Picture in picture)
#----Brave_Save_File
windowrulev2 = float,class:(brave),title:(Save File)
windowrulev2 = size 715 570,class:(brave),title:(Save File)
windowrulev2 = move 710 160,class:(brave),title:(Save File)

# Firefox
#--------2
#windowrule = workspace 2, ^(.*firefox.*)$
#windowrulev2 = float,class:(firefox),title:(Picture-in-Picture)
#windowrulev2 = size 600 350,class:(firefox),title:(Picture-in-Picture)
#windowrulev2 = move 1290 700,class:(firefox),title:(Picture-in-Picture)

# Code and Zed
#--------3
windowrule = workspace 3, ^(.*code-url-handler.*)$
windowrule = workspace 3, ^(.*dev.zed.Zed.*)$

# Thunar
#--------4
windowrule = workspace 4, ^(.*thunar.*)$
windowrulev2= size 400 150,class:(thunar),title:(Create New Folder)
windowrulev2= move 485 350,class:(thunar),title:(Create New Folder)
windowrulev2 = float,class:(thunar),title:(File Operation Progress)

#--------5

# Telegram and Discord
#--------6
windowrule = workspace 6, ^(.*org.telegram.desktop.*)$
windowrule = workspace 6, ^(.*discord.*)$
windowrule = workspace 6, ^(.*whatsapp-for-linux.*)$

# Celluloid, VLC, and G4Music
#--------7
windowrule = workspace 7, ^(.*io.github.celluloid_player.Celluloid.*)$
windowrule = workspace 7, ^(.*vlc.*)$
windowrule = workspace 7, ^(.*nuclear.*)$
windowrule = workspace 7, ^(.*com.github.neithern.g4music.*)$
windowrulev2 = float,class:(Player),title:(Player)

#--------8
windowrule = workspace 8, ^(.*uget-gtk.*)$

# Waydroid and Steam
#--------9
windowrule = workspace 9, ^(.*Waydroid.*)$
windowrule = workspace 9, ^(.*steam.*)$

#--------10
windowrule = workspace 10, ^(.*bleachbit.*)$

# Warp
windowrule=tile,^(dev.warp.Warp)$
windowrule=noblur,^(dev.warp.Warp)$

# Floatcritty
windowrule=float,^(floatcritty)$
windowrule=size 900 600,^(floatcritty)$
windowrule=move 510 50,^(floatcritty)$
windowrule=pin,^(floatcritty)$
windowrule=animation slide,^(floatcritty)$

# G4Music
windowrule=float,^(com.github.neithern.g4music)$

# Engrampa
windowrule=float,^(engrampa)$

# Kitty
windowrule=noblur,^(kitty)$
windowrule=float,^(kitty)
windowrule=size 900 520,^(kitty)
windowrule=move 230 160,^(kitty)

# MSEdge
#---BingGPT
windowrulev2 = size 606 1007 ,class:(Microsoft-edge),title:(BingGPT)
windowrulev2 = move 1314 52,class:(Microsoft-edge),title:(BingGPT)
windowrulev2 = animation slide,class:(Microsoft-edge),title:(BingGPT)

#---WhatsApp
windowrulev2 = workspace 6,class:(brave-hnpfjngllnobngcgfapefoaidbinmjnm-Default),title:(WhatsApp Web)
windowrulev2 = tile,class:(brave-hnpfjngllnobngcgfapefoaidbinmjnm-Default),title:(WhatsApp Web)

# NoBlur
windowrule=noblur,^(vlc)$
windowrule=noblur,^(brave-browser)$


Config File: /home/archkye/.config/hypr/default/layerrules.conf: Read Succeeded
layerrule = blur,rofi

#---AGS
layerrule = blur,bar-0

layerrule = ignorealpha 0.1, mpris
layerrule = blur, mpris

layerrule = ignorealpha 0.1,notifications
layerrule = blur,notifications

layerrule = ignorealpha 0.1,applauncher
layerrule = blur,applauncher

layerrule = ignorealpha 0.1,Chat
layerrule = blur,Chat

layerrule = ignorealpha 0.1,SideDash
layerrule = blur,SideDash


#---NoAnimations
# layerrule = animation ,Chat
# layerrule = animation ,SideDash
# layerrule = animation ,applauncher
# layerrule = animation ,volume_osd
# layerrule = noanim, mpris
layerrule = noanim, powermenu
layerrule = noanim, notifications

======Config-End========
```
</details>

I'm currently using hyprpicker-git from `chaotic-aur`, but the error persists with the "stable" of the arch's `EXTRA` repo.
This is the result of running it via terminal or keybinds:

<details>
    <summary>Video</summary>

https://github.com/user-attachments/assets/bab7720d-bb7b-44f9-86e2-e1be13dc3a48

</details>

I can't offer more information because I don't use hyprpicker often, so I can't pinpoint when this issue arose, but it must be recent.

I'm using every -git version of the hypr-ecosystem:

<details>
    <summary>Hypr-Ecosystem</summary>

```log
hyprcursor-git 0.1.9.r11.g912d5602-1
hypridle-git 0.1.2.r10.ge5366d34-1
hyprland-git 0.41.2.r165.2b520571-1
hyprlang-git 0.5.2.r8.gadbefbf4-1
hyprlock-git 0.4.1.r3.gcf0e975f-1
hyprpicker-git 0.3.0.r9.c36676a-1
hyprutils-git 0.2.1.r0.g5dcbbc1e-1
xdg-desktop-portal-hyprland-git 1.3.2.r4.gb9b97e5b-1

```

</details>

**Comment by vaxerski** [comment]
> I can't offer more information because I don't use hyprpicker often, so I can't pinpoint when this issue arose, but it must be recent.

try reverting everything to releases and see which git version breaks it

**Comment by ARKye03** [comment]
> try reverting everything to releases and see which git version breaks it

I can confirm that with:
<details>
    <summary>log</summary>
    
```log
hyprcursor 0.1.9-1
hypridle 0.1.2-1
hyprland 0.41.2-3
hyprlang 0.5.2-1
hyprlock 0.4.1-1
hyprpicker 0.3.0-1
hyprutils 0.2.0-1
hyprwayland-scanner-git 0.4.0.r4.ga048a6cb-1
xdg-desktop-portal-hyprland 1.3.3-2
```

</details>

Does not appear and hyprpicker acts as expected and is usable again. What else should I try?

**Comment by vaxerski** [comment]
well, get hyprland-git and test if it occurs due to hyprland, if so, bisect hyprland. If not, check hyprpicker

**Comment by ARKye03** [comment]
After learning how to `git bisect`, sorry for the delay, the result is:

<details>
<summary>Log</summary>

```git
37e1411e8d94fe8f3fb678588a7df9b8f931910f is the first bad commit
commit 37e1411e8d94fe8f3fb678588a7df9b8f931910f (HEAD)
Author: Vaxry <43317083+vaxerski@users.noreply.github.com>
Date:   Wed Jul 31 20:47:26 2024 +0100

    core/surface/buffer: Buffer lock/release fixes (#7110)

 src/desktop/LayerSurface.cpp         |   2 +-
 src/desktop/WLSurface.cpp            |  23 +++++++++++------------
 src/desktop/Window.cpp               |  10 +++++-----
 src/events/Windows.cpp               |   2 +-
 src/helpers/Monitor.cpp              |   7 +++----
 src/managers/PointerManager.cpp      |  50 +++++++++++++++++++++++++++++++++++++-------------
 src/protocols/DRMSyncobj.cpp         |   2 +-
 src/protocols/InputMethodV2.cpp      |   4 ++--
 src/protocols/LayerShell.cpp         |   2 +-
 src/protocols/SessionLock.cpp        |   2 +-
 src/protocols/Viewporter.cpp         |   4 ++--
 src/protocols/XDGShell.cpp           |   6 +++---
 src/protocols/core/Compositor.cpp    | 124 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-----------------------------------------------
 src/protocols/core/Compositor.hpp    |  35 ++++++++++++++++++++++-------------
 src/protocols/core/DataDevice.cpp    |   8 ++++----
 src/protocols/core/Seat.cpp          |  21 ++++++++++++++++++++-
 src/protocols/core/Seat.hpp          |  15 +++++++++++++++
 src/protocols/core/Subcompositor.cpp |   4 ++--
 src/protocols/types/Buffer.cpp       |  56 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 src/protocols/types/Buffer.hpp       |  28 +++++++++++++++++++++++++---
 src/protocols/types/SurfaceRole.hpp  |   1 +
 src/protocols/types/WLBuffer.cpp     |   1 -
 src/protocols/types/WLBuffer.hpp     |   2 --
 src/render/Renderer.cpp              |  14 +++++++-------
 src/render/Texture.cpp               |   3 +++
 src/render/Texture.hpp               |   1 +
 src/xwayland/XSurface.cpp            |   6 +++---
 27 files changed, 304 insertions(+), 129 deletions(-)
```

</details>

I'm on `548968279926a73d7ff19a9a185c977c50d56756` and hyprpicker works as intended
